server/src/main/java/org/opensearch/action/search/SearchCoordinatorStats.java

```diff
@@ -0,0 +1,169 @@
+    }
+    public long getExpandSearchPhaseStart() {
+        return expandSearchPhaseStart;
+    }
+    public long getExpandSearchPhaseEnd() {
+        return expandSearchPhaseEnd;
+    }
+
+    @Inject
+    public SearchCoordinatorStats () { }
+    public long getQueryMetric() {
+        return totalStats.queryMetric.sum();
+    }
+    public long getQueryCurrent() {
+        return totalStats.queryCurrent.count();
+    }
+    public long getQueryTotal() {
+        return totalStats.queryTotal.count();
+    }
+    public long getFetchMetric() {
+        return totalStats.fetchMetric.sum();
+    }
+    public long getFetchCurrent() {
+        return totalStats.fetchCurrent.count();
+    }
+    public long getFetchTotal() {
+        return totalStats.fetchTotal.count();
+    }
+    public long getExpandSearchMetric() {
+        return totalStats.expandSearchMetric.sum();
+    }
+    public long getExpandSearchCurrent() {
+        return totalStats.expandSearchCurrent.count();
+    }
+    public long getExpandSearchTotal() {
+        return totalStats.expandSearchTotal.count();
+    }
+    public void incQueryCurrent() { totalStats.queryCurrent.inc(); }
+    public void decQueryCurrent() { totalStats.queryCurrent.dec(); }
+    public void incQueryTotal() { totalStats.queryTotal.inc(); }
+    public void incFetchCurrent() { totalStats.fetchCurrent.inc(); }
+    public void decFetchCurrent() { totalStats.fetchCurrent.dec(); }
+    public void incFetchTotal() { totalStats.fetchTotal.inc(); }
+    public void incExpandSearchCurrent() { totalStats.expandSearchCurrent.inc(); }
+    public void decExpandSearch() { totalStats.expandSearchCurrent.dec(); }
+    public void incExpandSearch() { totalStats.expandSearchTotal.inc(); }
+
+    public void setStats(SearchCoordinatorStats searchCoordinatorStats) {
+        totalStats.queryMetric.inc(searchCoordinatorStats.totalStats.queryMetric.sum());
+        totalStats.queryCurrent.inc(searchCoordinatorStats.totalStats.queryCurrent.count());
+        totalStats.queryTotal.inc(searchCoordinatorStats.totalStats.queryTotal.count());
+        totalStats.fetchMetric.inc(searchCoordinatorStats.totalStats.fetchMetric.sum());
+        totalStats.fetchCurrent.inc(searchCoordinatorStats.totalStats.fetchCurrent.count());
+        totalStats.fetchTotal.inc(searchCoordinatorStats.totalStats.fetchTotal.count());
+        totalStats.expandSearchMetric.inc(searchCoordinatorStats.totalStats.expandSearchMetric.sum());
+        totalStats.expandSearchCurrent.inc(searchCoordinatorStats.totalStats.expandSearchCurrent.count());
+        totalStats.expandSearchTotal.inc(searchCoordinatorStats.totalStats.expandSearchTotal.count());
+    }
+    @Override
+    public void onQueryPhaseStart() {
+        totalStats.queryCurrent.inc();
+    }
+    @Override
+    public void onQueryPhaseEnd(long tookTime) {
+        totalStats.queryCurrent.dec();
+        totalStats.queryTotal.inc();
+        totalStats.queryMetric.inc(tookTime);
```

Comment:
Refer shard level stats logic for this. Lets use similar logic as it is thread safe (via computeStats method).